client.go

```diff
@@ -285,6 +285,36 @@ func (uc *Client) IsEnabled(feature string, options ...FeatureOption) (enabled b
 	return false
 }
 
+// GetVariant queries a variant as the specified feature is enabled.
+//
+// It is safe to call this method from multiple goroutines concurrently.
+func (uc *Client) GetVariant(feature string, fallbackVariant *api.Variant) *api.Variant {
```

Comment:
I don't think we should force the user to pass in the second argument here (even if it can be `nil`). To keep it consistent with the `IsEnabled` function I think you should change the signature to take a variadic option type like this:

```suggestion
func (uc *Client) GetVariant(feature string, options ...VariantOption) *api.Variant {
```
This also allows us to extend the API in the future without the need to add more arguments. The `VariantOption` type would be similar to the `FeatureOption` type which is here:

https://github.com/Unleash/unleash-client-go/blob/v3/config.go#L142

If you are interested in learning more about this pattern, Dave Cheney has a great talk here:

https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis